### ISSUE TITLE
Image processing library for QPython3

### ISSUE THREAD

**Issue by micechal** [qpython-android/qpython.org#51]
Hi. I've been looking for a library that'd allow to for example resize some image to given width and height.
I've found a couple of such libraries, but unfortunately none of them installed properly with "pip install".
You can see the logs here:
Pillow: http://pastebin.ca/2966039
PIL: http://pastebin.ca/2966053
SciPy: http://pastebin.ca/2966055
Scikit-image: http://pastebin.ca/2966056
And after installing "six" (successful) it requires "matplotlib", which fails: http://pastebin.ca/2966060

It would be really great if any of these libraries was made to work somehow, allowing for basic tasks, such as resizing or cropping images.

**Comment by riverfor** [comment]
Thanks for detailed feedback.

QPython or QPython3 can not cross-compile the files which were implemented with c or c++. So the pip install processing failed. It's a big challenge to implement the toolchain for qpython on android, but I will try to do that in the future.

And QPython has included the PIL library by default. Is it ok for you?

**Comment by micechal** [comment]
By QPython you mean the one without '3', right? I tried importing PIL in QPython3 and it can't find it: `ImportError: No module named PIL`

**Comment by riverfor** [comment]
Yes, there are two branches for android python, one is QPython, which
includes the python 2 core, the other is QPython3, which includes the
python 3 core.

QPython has integrated the PIL module by default.

On Tue, Mar 31, 2015 at 2:11 PM, Michał Durak <notifications@github.com>
wrote:

> By QPython you mean the one without '3', right? I tried importing PIL in
> QPython3 and it can't find it: ImportError: No module named PIL
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/qpython-android/qpython.org/issues/51#issuecomment-87955097>
> .
>



-- 
river's showtime, river's game

**Comment by micechal** [comment]
I've tried QPython and PIL is indeed there by default and it works,  but it feels weird having to switch to that legacy language syntax and whatnot. I think it would be reasonable to have an equivalent library in QPython3. I'd be happy to try and compile the module myself using my computer and provide the binaries somewhere for others, it's just that I couldn't really find any guide of sorts describing the cross-compilling process. All I was able to find was a couple of dead/inexistent wiki pages.

**Comment by tomoemon** [comment]
Are there any updates?
QPYPI on QPython3 still doesn't seem to include Pillow.
